python/ray/tune/integration/mlflow.py

```
@@ -94,55 +86,25 @@ def __init__(
-            # First check if experiment with name exists.
-            experiment = self.client.get_experiment_by_name(
-                self.experiment_name)
-            if experiment is not None:
-                # If it already exists then get the id.
-                experiment_id = experiment.experiment_id
-            else:
-                # If it does not exist, create the experiment.
-                experiment_id = self.client.create_experiment(
-                    name=self.experiment_name)
-        else:
-            # No experiment_name is passed in and name env var is not set.
-            # Now check the experiment id env var.
-            experiment_id = os.environ.get("MLFLOW_EXPERIMENT_ID")
-            # Confirm that an experiment with this id exists.
-            if experiment_id is None or self.client.get_experiment(
-                    experiment_id) is None:
-                raise ValueError("No experiment_name passed, "
-                                 "MLFLOW_EXPERIMENT_NAME env var is not "
-                                 "set, and MLFLOW_EXPERIMENT_ID either "
-                                 "is not set or does not exist. Please "
-                                 "set one of these to use the "
-                                 "MLflowLoggerCallback.")
+    def setup(self, *args, **kwargs):
+        # Setup the mlflow logging util.
+        success = self.mlflow_util.setup_mlflow(
+            tracking_uri=self.tracking_uri,
+            registry_uri=self.registry_uri,
+            experiment_name=self.experiment_name)
+
+        if not success:
+            raise ValueError("No experiment_name passed, "
+                             "MLFLOW_EXPERIMENT_NAME env var is not "
+                             "set, and MLFLOW_EXPERIMENT_ID either "
+                             "is not set or does not exist. Please "
+                             "set one of these to use the "
+                             "MLflowLoggerCallback.")
```

See Matt's above comment, let's raise these errors in the mlflow util.

---

Addressed!